tf/edgeml/graph/bonsai.py

```diff
@@ -1,167 +1,213 @@
-        V_ = self.V[0:(self.numClasses)]
-
-        self.__nodeProb = []
-        self.__nodeProb.append(1)
-
-        score_ = self.__nodeProb[0] * tf.multiply(
-            tf.matmul(W_, X_), tf.tanh(self.sigma * tf.matmul(V_, X_)))
-        for i in range(1, self.totalNodes):
-            W_ = self.W[i * self.numClasses:((i + 1) * self.numClasses)]
-            V_ = self.V[i * self.numClasses:((i + 1) * self.numClasses)]
-
-            T_ = tf.reshape(self.T[int(np.ceil(i / 2.0) - 1.0)],
-                            [-1, self.projectionDimension])
-            prob = (1 + ((-1)**(i + 1)) *
-                    tf.tanh(tf.multiply(sigmaI, tf.matmul(T_, X_))))
-
-            prob = tf.divide(prob, 2.0)
-            prob = self.__nodeProb[int(np.ceil(i / 2.0) - 1.0)] * prob
-            self.__nodeProb.append(prob)
-            score_ += self.__nodeProb[i] * tf.multiply(
-                tf.matmul(W_, X_), tf.tanh(self.sigma * tf.matmul(V_, X_)))
-
-        self.score = score_
-        self.X_ = X_
-        return self.score, self.X_
-
-    def getPrediction(self):
-        '''
-        Takes in a score tensor and outputs a integer class for each data point
-        '''
-
-        if self.prediction is not None:
-            return self.prediction
-
-        if self.numClasses > 2:
-            self.prediction = tf.argmax(tf.transpose(self.score), 1)
-        else:
-            self.prediction = tf.argmax(
-                tf.concat([tf.transpose(self.score),
-                           0 * tf.transpose(self.score)], 1), 1)
-
-        return self.prediction
-
-    def assertInit(self):
-        errRank = "All Parameters must has only two dimensions shape = [a, b]"
-        assert len(self.W.shape) == len(self.Z.shape), errRank
-        assert len(self.W.shape) == len(self.T.shape), errRank
-        assert len(self.W.shape) == 2, errRank
-        msg = "W and V should be of same Dimensions"
-        assert self.W.shape == self.V.shape, msg
-        errW = "W and V are [numClasses*totalNodes, projectionDimension]"
-        assert self.W.shape[0] == self.numClasses * self.totalNodes, errW
-        assert self.W.shape[1] == self.projectionDimension, errW
-        errZ = "Z is [projectionDimension, dataDimension]"
-        assert self.Z.shape[0] == self.projectionDimension, errZ
-        assert self.Z.shape[1] == self.dataDimension, errZ
-        errT = "T is [internalNodes, projectionDimension]"
-        assert self.T.shape[0] == self.internalNodes, errT
-        assert self.T.shape[1] == self.projectionDimension, errT
-        assert int(self.numClasses) > 0, "numClasses should be > 1"
-        msg = "# of features in data should be > 0"
-        assert int(self.dataDimension) > 0, msg
-        msg = "Projection should be  > 0 dims"
-        assert int(self.projectionDimension) > 0, msg
-        msg = "treeDepth should be >= 0"
-        assert int(self.treeDepth) >= 0, msg
+	def __init__(self, numClasses, dataDimension, projectionDimension,
```

Comment:
@siddharthdivi Please change all tabs to spaces. The entire file is being shown as a diff.